### ISSUE TITLE
Disable window transparency when DWM is not enabled

### ISSUE THREAD

**Issue by Ivshti** [atom/electron#3155]
Ever since Electron has been based on Chrome 45, enabling transparent on a BrowserWindow in an environment where it can't be initialized - e.g. Windows 7 without aero enabled - causes the window to be practically invisible but still created. The window is still active, takes clicks, initializes the DOM, even shows the tooltips if you hover on to something, but it **does't draw at all**. Screenshot is not applicable, since the window is completely invisible. It also retains artifacts of other windows that have drawn on top of it.

This happens on Windows environments where transparency **cannot be initialized** (no aero) with Electron 0.29 and it **just starts in a non-transparent window**.

In other words, the graceful degradation to a non-transparent window is broken.

May be something related to window styles or how transparency not initializing is handled.

The way this was resolved in NW.js and previous versions of electron is checking ``DwmCompositionEnabled`` and then making the adjustments,

**Comment by zcbenz** [comment]
I'm :+1: on disabling transparency when DWM is not available, probably also disable GPU acceleration as per #1821.

**Comment by Ivshti** [comment]
@zcbenz do you have any idea what caused this? I remember in NW.js and maybe older electron versions - there was code to initialize transparency in something like win/browser_window.cc (??) that checked those things and added/removed windows styles.

**Comment by zcbenz** [comment]
I think Chromium team simply never tested the transparency code on machines with DWM disabled, it is an experiment feature and not being actively worked on. So as Chrome's GPU acceleration getting heavily changed, it is very easy to break things.

**Comment by zcbenz** [comment]
In Chromium window transparency is only enabled for Windows when DWM is enabled, so if we want to ensure window transparency works perfectly we have to disable it on most platforms.

Since now we have API to check whether DWM is enabled, I think it is better to put the choice of whether to enable window transparency to the users.
